src/ext/ddtrace.c

```diff
@@ -795,13 +807,169 @@ static PHP_FUNCTION(dd_tracer_circuit_breaker_info) {
 }
 
 #if PHP_VERSION_ID < 70000
-typedef int ddtrace_zppstrlen_t;
 typedef long ddtrace_zpplong_t;
 #else
-typedef size_t ddtrace_zppstrlen_t;
 typedef zend_long ddtrace_zpplong_t;
 #endif
 
+static PHP_FUNCTION(ddtrace_config_app_name) {
+    PHP5_UNUSED(return_value_used, this_ptr, return_value_ptr, ht);
+    ddtrace_string default_str = {
+        .ptr = NULL,
+        .len = 0,
+    };
+#if PHP_VERSION_ID < 70000
+    if (zend_parse_parameters(ZEND_NUM_ARGS() TSRMLS_CC, "|s", &default_str.ptr, &default_str.len) != SUCCESS) {
+        RETURN_NULL()
+    }
+#else
+    zend_string *default_zstr = NULL;
+    if (zend_parse_parameters(ZEND_NUM_ARGS() TSRMLS_CC, "|S", &default_zstr) != SUCCESS) {
+        RETURN_NULL()
+    }
+    if (default_zstr) {
+        default_str.ptr = ZSTR_VAL(default_zstr);
+        default_str.len = ZSTR_LEN(default_zstr);
+    }
+#endif
+
+    ddtrace_string service_name = {
+        .ptr = "DD_SERVICE_NAME",
+        .len = sizeof("DD_SERVICE_NAME") - 1,
+    };
+    ddtrace_string app_name;
+    char *value = getenv(service_name.ptr);
```

Comment:
For all the `getenv()` calls, is there a reason you did not use the existing configuration API (`get_dd_trace_*()`)? Since it memoizes the `getenv()` calls, I'm wondering if it should be considered.

Reply:
Actually, I avoided it _because_ it caches the calls. It breaks a lot of tests which just set the env and then expect it to work.

The main reason it does "caching" is to make it thread-safe so our background sender can read this info. Because of this, there are locks involved for reading the values so it's not a particularly "cheap" cache.

At least for now I thought it would be better to avoid code churn. If we get even more performance advantage then we can measure and do that later.